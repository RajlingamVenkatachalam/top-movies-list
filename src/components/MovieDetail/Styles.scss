@media screen and (max-width: 1500px) {
    .card {
        width: 30%;
        height: 90%;
        margin: 10px;
    }
}

@media screen and (max-width: 600px) {
    .card {
        width: 100%;
        height: fit-content;
        margin: 10px;
    }
}

.media {
    width: 250px;
    height: 250px;
    margin: 10px;
}