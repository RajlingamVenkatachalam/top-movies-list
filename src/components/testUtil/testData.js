export const orderSelect = [{
        "label": "Release Date",
        "valueToOrderBy": "releaseDate"
      }, {
        "label": "Rank",
        "valueToOrderBy": "rank"
      }];

export const moviesList = [
    {
        "id": 2,
        "type": "poster",
        "rank": 2,
        "synopsis": "Epic tale in which an intrepid archaeologist tries to beat a band of Nazis to a unique religious relic which is central to their plans for world domination. Battling against a snake phobia and a vengeful ex-girlfriend, Indiana Jones is in constant peril, making hair's-breadth escapes at every turn in this celebration of the innocent adventure movies of an earlier era.",
        "title": "Raiders of the Lost Ark",
        "imageUrl": "https://preview.ibb.co/fn5Xyp/raiders.jpg",
        "releaseDate": 1981
      }, {
        "id": 4,
        "type": "poster",
        "rank": 1,
        "synopsis": "Deckard (Harrison Ford) is forced by the police Boss (M. Emmet Walsh) to continue his old job as Replicant Hunter. His assignment: eliminate four escaped Replicants from the colonies who have returned to Earth. Before starting the job, Deckard goes to the Tyrell Corporation and he meets Rachel (Sean Young), a Replicant girl he falls in love with.",
        "title": "Blade Runner",
        "imageUrl": "https://preview.ibb.co/h9434U/blade_Runner.jpg",
        "releaseDate": 1982
      }, {
        "id": 3,
        "type": "poster",
        "rank": 4,
        "synopsis": "Disguised as a human, a cyborg assassin known as a Terminator (Arnold Schwarzenegger) travels from 2029 to 1984 to kill Sarah Connor (Linda Hamilton). Sent to protect Sarah is Kyle Reese (Michael Biehn), who divulges the coming of Skynet, an artificial intelligence system that will spark a nuclear holocaust. Sarah is targeted because Skynet knows that her unborn son will lead the fight against them. With the virtually unstoppable Terminator in hot pursuit, she and Kyle attempt to escape.",
        "title": "The Terminator",
        "imageUrl": "https://preview.ibb.co/caykJp/terminator.jpg",
        "releaseDate": 1984
      }
];
