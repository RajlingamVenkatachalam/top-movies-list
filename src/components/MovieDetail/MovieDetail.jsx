import React from "react";
import PropTypes from "prop-type";
import { CardContent, Typography } from "@material-ui/core";
import "./Styles.scss";

const OtherDetailView = ({ label, value }) =>{
    return (
        <>
            <Typography variant="h6" component="h2">
                {label}
            </Typography>                         
            <Typography color="textSecondary">
                {value}
            </Typography>
        </>
    )
}


const MovieDetail = ({ movieDetail }) => {
    return(
        <div className="card">
            <img
                className="media"
                src={movieDetail.imageUrl}
                title={movieDetail.title}
            />
            <CardContent>
                <Typography  gutterBottom variant="h5" component="h2">
                    {movieDetail.title}
                </Typography>
                <Typography variant="body2" color="textSecondary" component="p">
                    {movieDetail.synopsis}
                </Typography>
                <div style={{textAlign:"left", marginTop:"10px"}}>
                    <OtherDetailView label="Release Date" value={movieDetail.releaseDate} />
                    <OtherDetailView label="Rank" value={movieDetail.rank} />
                </div>                       
            </CardContent>
        </div>
    )
}

MovieDetail.propTypes = {
    movieDetail: PropTypes.object
}

export default MovieDetail;