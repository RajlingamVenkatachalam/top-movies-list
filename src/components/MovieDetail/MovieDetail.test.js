import React from "react";
import renderer from "react-test-renderer";
import MovieDetail from "./MovieDetail";
import { moviesList } from "../testUtil/testData";

describe("test MovieDetail", () => {
    it("should display the Detail without any crash", () =>{
        let component = renderer.create(<MovieDetail movieDetail={moviesList[0]} />);
        expect(component.toJSON()).toMatchSnapshot();
    });
})