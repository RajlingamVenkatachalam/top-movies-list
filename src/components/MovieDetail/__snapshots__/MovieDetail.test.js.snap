// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test MovieDetail should display the Detail without any crash 1`] = `
<div
  className="card"
>
  <img
    className="media"
    src="https://preview.ibb.co/fn5Xyp/raiders.jpg"
    title="Raiders of the Lost Ark"
  />
  <div
    className="MuiCardContent-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
    >
      Raiders of the Lost Ark
    </h2>
    <p
      className="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
    >
      Epic tale in which an intrepid archaeologist tries to beat a band of Nazis to a unique religious relic which is central to their plans for world domination. Battling against a snake phobia and a vengeful ex-girlfriend, Indiana Jones is in constant peril, making hair's-breadth escapes at every turn in this celebration of the innocent adventure movies of an earlier era.
    </p>
    <div
      style={
        Object {
          "marginTop": "10px",
          "textAlign": "left",
        }
      }
    >
      <h2
        className="MuiTypography-root MuiTypography-h6"
      >
        Release Date
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        1981
      </p>
      <h2
        className="MuiTypography-root MuiTypography-h6"
      >
        Rank
      </h2>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextSecondary"
      >
        2
      </p>
    </div>
  </div>
</div>
`;
